pkgs/development/python-modules/keras/default.nix

```diff
@@ -1,39 +1,145 @@
   };
 
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  propagatedBuildInputs = [
+    absl-py
+    dm-tree
+    h5py
+    markdown-it-py
+    namex
+    numpy
+    # requests
+    rich
+    # scipy
+  ];
+
+  # passthru.optional-dependencies = {
+  #   cli = [
+  #     click
+  #     click-default-group
+  #     docformatter
+  #     jinja2
+  #     toposort
+  #   ];
+  #   lxml = [
+  #     lxml
+  #   ];
+  #   soap = [
+  #     requests
+  #   ];
+  # };
+
+  pythonImportsCheck = [
+    "keras"
+    # "keras.api"
+  ];
+
   nativeCheckInputs = [
-    pytest
-    pytest-cov
+    # pytest-cov
+    jax
+    jaxlib
+    pandas
     pytest-xdist
+    pytestCheckHook
+    tensorflow
+    torch
   ];
 
-  propagatedBuildInputs = [
-    six pyyaml numpy scipy h5py
-    keras-applications keras-preprocessing
+  preCheck = ''
+    export TMP=$(mktemp -d)
+    export HOME=$(mktemp -d)
+  '';
+
+  disabledTests = [
+    # Require internet access
+    "test_application_base_ConvNeXtLarge_channels_last"
+    "test_application_base_DenseNet201_channels_last"
+    "test_application_base_EfficientNetB0_channels_last"
+    "test_application_base_EfficientNetB1_channels_last"
+    "test_application_base_EfficientNetB2_channels_last"
+    "test_application_base_EfficientNetB3_channels_last"
+    "test_application_base_EfficientNetB4_channels_last"
+    "test_application_base_EfficientNetB5_channels_last"
+    "test_application_base_EfficientNetB6_channels_last"
+    "test_application_base_EfficientNetB7_channels_last"
+    "test_application_base_EfficientNetV2B0_channels_last"
+    "test_application_base_EfficientNetV2B1_channels_last"
+    "test_application_base_EfficientNetV2B2_channels_last"
+    "test_application_base_EfficientNetV2B3_channels_last"
+    "test_application_base_EfficientNetV2L_channels_last"
+    "test_application_base_EfficientNetV2M_channels_last"
+    "test_application_base_EfficientNetV2S_channels_last"
+    "test_application_base_MobileNet_channels_last"
+    "test_application_base_MobileNetV2_channels_last"
+    "test_application_base_MobileNetV3Large_channels_last"
+    "test_application_base_MobileNetV3Small_channels_last"
+    "test_application_base_NASNetLarge_channels_last"
+    "test_application_base_NASNetMobile_channels_last"
+    "test_application_base_InceptionResNetV2_channels_last"
+    "test_application_base_InceptionV3_channels_last"
+    "test_application_base_ResNet101_channels_last"
+    "test_application_base_ResNet101V2_channels_last"
+    "test_application_base_ResNet152_channels_last"
+    "test_application_base_ResNet152V2_channels_last"
+    "test_application_base_ResNet50_channels_last"
+    "test_application_base_ResNet50V2_channels_last"
+    "test_application_base_VGG16_channels_last"
+    "test_application_base_VGG19_channels_last"
+    "test_application_base_Xception_channels_last"
   ];
 
-  # Couldn't get tests working
+  # Multiple package conflicts
```

Comment:
What does this mean exactly?

Reply:
The `catchConflicts` (or similar) hook catches a bunch of conflicting versions.

Reply:
If I set `doCheck = true;` in the main derivation, I get:
```
Running phase: pythonCatchConflictsPhase
Found duplicated packages in closure for dependency 'grpcio': 
        grpcio 1.60.0 (/nix/store/f7ycslpfvcf3kafiyz3fwpq3wblkwvgm-python3.11-grpcio-1.60.0/lib/python3.11/site-packages/grpcio-1.60.0.dist-info)
        grpcio 1.60.0 (/nix/store/74pwwrlya6pzxvawvrjwhn5gjcd41pbd-python3.11-grpcio-1.60.0/lib/python3.11/site-packages/grpcio-1.60.0.dist-info)
Found duplicated packages in closure for dependency 'protobuf': 
        protobuf 4.21.12 (/nix/store/fpcpf3fs5dbn0g3inkm2gl1zxk13d123-python3.11-protobuf-4.21.12/lib/python3.11/site-packages/protobuf-4.21.12.dist-info)
        protobuf 4.24.4 (/nix/store/gknbg49nip7j60a53ch5g1b0hbxvqrhi-python3.11-protobuf-4.24.4/lib/python3.11/site-packages/protobuf-4.24.4.dist-info)
Found duplicated packages in closure for dependency 'tensorboard': 
        tensorboard 2.15.1 (/nix/store/nc14v4y90qq0dnkd7w0h2z0qww8n2a09-python3.11-tensorboard-2.15.1/lib/python3.11/site-packages/tensorboard-2.15.1.dist-info)
        tensorboard 2.15.1 (/nix/store/5k9zm168mdh2myiqhjzk3g66s6d5d7gc-python3.11-tensorboard-2.15.1/lib/python3.11/site-packages/tensorboard-2.15.1.dist-info)
Found duplicated packages in closure for dependency 'tensorboard_plugin_profile': 
        tensorboard_plugin_profile 2.11.1 (/nix/store/fbz0xbc0ld6fyj98a94clqy23qiw2a05-python3.11-tensorboard_plugin_profile-2.11.1/lib/python3.11/site-packages/tensorboard_plugin_profile-2.11.1.dist-info)
        tensorboard_plugin_profile 2.11.1 (/nix/store/x5z9hg5z0yfwbbxbhp43qkx938k0szrd-python3.11-tensorboard_plugin_profile-2.11.1/lib/python3.11/site-packages/tensorboard_plugin_profile-2.11.1.dist-info)
```

Reply:
It's the tensorflow protobuf issue again, right? Did we open anything yet to track it specifically and to reference from other places?

Reply:
> It's the tensorflow protobuf issue again, right?

Yes, but not alone :/